### ISSUE TITLE
Question on alignment depth used for SiT-B/2

### ISSUE THREAD

**Issue by dlsrbgg33** [sihyun-yu/REPA#5]
Thank you for your excellent work.

I have a question regarding the alignment depth used for SiT-B/2. According to Table 6, it appears that the alignment depth is 8 for SiT-B/2, but in Table 7, it seems to indicate that the base model uses an alignment depth of 4. Could you clarify if I'm misunderstanding something?

**Comment by sihyun-yu** [comment]
Hi! We used layer 4 for SiT-B/2 --- I accidentally swapped the alignment depth for Figure 3 and Table 3 (SiT-B). Thanks for pointing this out and we will fix in the revision :)

**Comment by dlsrbgg33** [comment]
Thank you!
To make it clear, layer 4 is for SiT-B/2 while bigger models use layer 8 for alignment, correct?

**Comment by sihyun-yu** [comment]
Exactly!

**Comment by dlsrbgg33** [comment]
I tried to reproduce SiT-B/2 model with alignment depth 4, but I got 23 FID which 1.4 FID better than paper score.
I have not used cfg.

below is my training script:
`accelerate launch 
  --mixed-precision="fp16" 
  --multi-gpu --num_machines=1 --num_processes=8 --gpu_ids=0,1,2,3,4,5,6,7 
  train.py 
  --allow-tf32 
  --mixed-precision="fp16" 
  --seed=0 
  --path-type="linear" 
  --prediction="v" 
  --weighting="uniform" 
  --model="SiT-B/2" 
  --enc-type="dinov2-vit-b" 
  --proj-coeff=0.5 
  --encoder-depth=4 
  --output-dir="exps" 
  --exp-name="linear-sitb-dinov2-b-enc4"`

below is my evaluation script:
`torchrun --nnodes=1 --nproc_per_node=8  generate.py 
  --model SiT-B/2 
  --num-fid-samples 50000 
  --ckpt exps/linear-sitb-dinov2-b-enc4/checkpoints/0400000.pt 
  --path-type=linear 
  --encoder-depth=4 
  --projector-embed-dims=768 
  --per-proc-batch-size=64 
  --mode=sde 
  --num-steps=250 
  --cfg-scale=1.0 `

The reason why I got different score is related to setting legacy in [https://github.com/sihyun-yu/REPA/blob/9d60489e823bd9b356319b6d30e4cf7f333119a9/train.py#L283C13-L283C22](url)?

Thank you in advance.

**Comment by sihyun-yu** [comment]
Thanks for sharing your result. Yes it is probably related to legacy option --- the original code drops out the class label slightly more (which is unintended), FID can be better by adjusting this issue, because it will let the model trained more with image-label pair without label dropout.

I will keep this legacy option for exact reproducibility in the result in the paper. Thanks again for checking this! You will have a similar result in the paper if you add --legacy option.
